### ISSUE TITLE
[Bug] C++ code issue with Ids assigned to variables [sf#186]

### ISSUE THREAD

**Issue by DietmarSchwertberger** [DietmarSchwertberger/wxGlade#261]
**Reported by ggbsf on 2016-04-22 14:45 UTC**
There is a bug with c++ code generation, related to item ids.
I have to use the following fix at line 991 of cpp_codegen.y:
~~~python
		#LB bugfix
        for l in init_lines:
			if l.endswith("\n"):
				swrite(tab + l)
~~~

**Comment by DietmarSchwertberger** [comment]
**Commented by cgrohmann on 2016-04-22 19:01 UTC**
- Description has changed:

Diff:

~~~~

--- old
+++ new
@@ -1,6 +1,8 @@
 There is a bug with c++ code generation, related to item ids.
 I have to use the following fix at line 991 of cpp_codegen.y:
+~~~python
 		#LB bugfix
         for l in init_lines:
 			if l.endswith("\n"):
 				swrite(tab + l)
+~~~

~~~~

- **assigned_to**: Carsten Grohmann
- **Affected Version**:  --> 0.7.2

**Comment by DietmarSchwertberger** [comment]
**Commented by cgrohmann on 2016-04-22 19:01 UTC**
Hi Luigi,

please describe the issue a little bit more detailed. It would be really great if you could add a list of steps to reproduce the unexpected and the expected behaviour.

That information allows me to add a simple test to prevent later regressions.

Thank you,
Carsten

**Comment by DietmarSchwertberger** [comment]
**Commented by ggbsf on 2016-04-27 13:06 UTC**
That happens specifying an id for an item in the properties in the form: <myid>=?.
The .h with the enum is generated ok, but in the .cpp when the items are created the code is confused as if it contains also the declaration of the enum and doesn't compile.
I temporarly solved with that line added in the middle of the code generation (if l.endswith("\n")).

**Comment by DietmarSchwertberger** [comment]
**Updated by cgrohmann on 2016-05-24 16:47 UTC**
- **summary**: c++ code generation issue with item ids --> C++ code issue with Ids assigned to variables
- **status**: open --> closed

**Comment by DietmarSchwertberger** [comment]
**Commented by cgrohmann on 2016-05-24 16:47 UTC**
Hi Luigi,

the bug is fixed in revision 1939:9ab49f35693a "Fix C++ code issue with Ids assigned to variables (sf bug #186)" of the current development branch as well as in revision 1940:9b44ee204739 of the current stable branch. 

The fix will be part of the next official release.

You can download and use the fixed version from the stable branch directly: https://bitbucket.org/agriggio/wxglade/get/WXG-BRANCH-VERSION_0_7_2.tar.gz .

Alternatively you can download the currend development branch from https://bitbucket.org/agriggio/wxglade/get/default.tar.gz .

The installation from such a downloaded source is described in section "Installation" of the wxGlade manual at http://wxglade.sourceforge.net/manual/ch02s02.html.

Thank you for reporting this bug.
Carsten
